### PR TITLE
Adjust the North Carolina API

### PR DESCRIPTION
**Pull request by jbradberry**
to break out the Hispanic ethnicity choice into a race category, with
totals removed from the remainder of the race category that they
originally fell under.

### CODE REVIEW

File: nc/api.py [hunk 1]
```diff
@@ -77,22 +75,26 @@ def query(self, results, group_by, filter_=None):
...
-                data[race] = stop['count']
-            if 'person__ethnicity' in group_by:
-                ethnicity = GROUPS.get(stop['person__ethnicity'],
-                                       stop['person__ethnicity'])
-                data[ethnicity] = stop['count']
+                # The 'Hispanic' ethnicity option is now being aggreggated into its
+                # own race category, and its count excluded from the other counts.
+                if stop['person__ethnicity'] == 'H':
+                    race = GROUPS.get('H', 'H')
+                elif 'person__race' in group_by:
```
Comment (dchukhin) [inline nc/api.py]: It seems redundant to check for 'person__race' being in `group_by` for a second time here, no?

File: nc/tests/test_api.py [hunk 2]
```diff
@@ -23,7 +23,7 @@ def test_list_agencies(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         # Other Agencies may have been left around from other tests
         self.assertIn((agency.pk, agency.name), [
-            (a.pk, a.name) for a in Agency.objects.all()
+            (a['id'], a['name']) for a in response.data
         ])
```
Comment (dchukhin) [inline nc/tests/test_api.py]: Can we add a comment, maybe at the top of this test case, that the api treats "Hispanic" as a race, even though it's stored in the `ethnicity` field in the database?

File: nc/tests/test_api.py [hunk 3]
```diff
@@ -193,7 +209,8 @@ def test_contraband_hit_rate(self):
 
         searches = response.data['searches']
         self.assertEqual(searches[0]['year'], 2010)
-        self.assertEqual(searches[0][race_label], 2)
+        self.assertEqual(searches[0][race_label], 1)
+        self.assertEqual(searches[0]['hispanic'], 1)
         self.assertEqual(searches[1]['year'], 2012)
         self.assertEqual(searches[1][race_label], 1)
```
Comment (dchukhin) [inline nc/tests/test_api.py]: Can we also add a few lines below these testing `response.data['contraband']`?

Comment (jbradberry) [reply nc/tests/test_api.py]: That is already there, lines 218 - 221.

Comment (dchukhin) [reply nc/tests/test_api.py]: Good point

**Pull request merged**